source/slang/slang-ast-type.cpp

```diff
@@ -77,6 +77,11 @@ Type* Type::getCanonicalType()
 
 // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! OverloadGroupType !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
+void OverloadGroupType::_hashIntoOverride(DigestBuilder& builder)
+{
+    builder.addToDigest(_getHashCodeOverride());
```

Comment:
Can't just use the existing hash code. This hash code has high chance of collision so they are not safe to use. This implementation should recurse down the same way as `equalsValOverride`.